### PR TITLE
VZ-7975: Check for scrape targets instead of every metric in tests

### PR DESCRIPTION
**Pull request by IshaGirdhar**
Most of the end-to-end tests that validate metrics do so by querying Prometheus for various metrics to ensure they have been collected. It should be sufficient for this type of test to verify that Prometheus is able to scrape the expected endpoints. This PR replaces the logic from checking each metric to just checking the scrape target endpoints' health.

### CODE REVIEW

File: tests/e2e/examples/bobsbooks/bobs_books_test.go [hunk 1]
```diff
@@ -261,63 +277,21 @@ var _ = t.Describe("Bobs Books test", Label("f:app-lcm.oam",
 		})
 	})
 	t.Context("Metrics.", Label("f:observability.monitoring.prom"), FlakeAttempts(5), func() {
-		// Verify application Prometheus scraped metrics
+
+		kubeconfigPath, err := k8sutil.GetKubeConfigLocation()
```
Comment (kminder) [inline tests/e2e/examples/bobsbooks/bobs_books_test.go:281]: Seems like this should be done int he before suite that that it doesn't generate a bunch of meaningless errors if the kubeconfig can't be found.

File: tests/e2e/examples/bobsbooks/bobs_books_test.go [hunk 2]
```diff
@@ -261,63 +277,21 @@ var _ = t.Describe("Bobs Books test", Label("f:app-lcm.oam",
...
+		if err != nil {
+			Expect(err).To(BeNil(), fmt.Sprintf("Failed to get default kubeconfig path: %s", err.Error()))
+		}
+		ok, _ := pkg.IsVerrazzanoMinVersion("1.4.0", kubeconfigPath)
```
Comment (kminder) [inline tests/e2e/examples/bobsbooks/bobs_books_test.go:285]: Please use a better variable name than 'ok'.
Should this also be done in the before suite?
The error should not be ignored.

**Review by kminder — APPROVED** [review]
Approved.

**Comment by IshaGirdhar** [comment]
> 

@kminder Lost approval because of master rebasing. Could you approve it again?

**Review by kminder — APPROVED** [review]
Approved.

**Pull request merged**